sklearn/datasets/tests/test_kddcup99.py

```diff
@@ -6,41 +6,62 @@
 is too big to use in unit-testing.
 """
 
-from sklearn.datasets.tests.test_common import check_return_X_y
 from functools import partial
+import pytest
+
+from sklearn.datasets import fetch_kddcup99
+from sklearn.datasets.tests.test_common import check_as_frame
+from sklearn.datasets.tests.test_common import check_pandas_dependency_message
+from sklearn.datasets.tests.test_common import check_return_X_y
+
 
+@pytest.mark.parametrize("as_frame", [True, False])
+@pytest.mark.parametrize(
+    "subset, n_samples, n_features",
+    [(None, 494021, 41),
+     ("SA", 100655, 41),
+     ("SF", 73237, 4),
+     ("http", 58725, 3),
+     ("smtp", 9571, 3)]
+)
+def test_fetch_kddcup99_percent10(
+    fetch_kddcup99_fxt, as_frame, subset, n_samples, n_features
+):
+    data = fetch_kddcup99_fxt(subset=subset, as_frame=as_frame)
+    assert data.data.shape == (n_samples, n_features)
+    assert data.target.shape == (n_samples,)
+    if as_frame:
+        assert data.frame.shape == (n_samples, n_features + 1)
 
-def test_percent10(fetch_kddcup99_fxt):
-    data = fetch_kddcup99_fxt()
 
-    assert data.data.shape == (494021, 41)
-    assert data.target.shape == (494021,)
+def test_fetch_kddcup99_return_X_y(fetch_kddcup99_fxt):
+    fetch_func = partial(fetch_kddcup99_fxt, subset='smtp')
+    data = fetch_func()
+    check_return_X_y(data, fetch_func)
 
-    data_shuffled = fetch_kddcup99_fxt(shuffle=True, random_state=0)
-    assert data.data.shape == data_shuffled.data.shape
-    assert data.target.shape == data_shuffled.target.shape
 
-    data = fetch_kddcup99_fxt(subset='SA')
-    assert data.data.shape == (100655, 41)
-    assert data.target.shape == (100655,)
+def test_fetch_kddcup99_as_frame(fetch_kddcup99_fxt):
+    bunch = fetch_kddcup99_fxt()
+    check_as_frame(bunch, fetch_kddcup99_fxt)
 
-    data = fetch_kddcup99_fxt(subset='SF')
-    assert data.data.shape == (73237, 4)
-    assert data.target.shape == (73237,)
 
-    data = fetch_kddcup99_fxt(subset='http')
-    assert data.data.shape == (58725, 3)
-    assert data.target.shape == (58725,)
+def test_fetch_kddcup99_shuffle(fetch_kddcup99_fxt):
+    dataset = fetch_kddcup99_fxt(
+        random_state=0, subset='SA', percent10=True,
+    )
+    dataset_shuffled = fetch_kddcup99_fxt(
+        random_state=0, subset='SA', shuffle=True, percent10=True,
+    )
+    assert any(dataset_shuffled.target[-100:] == b'normal.')
```

Comment:
If this is testing shuffle should we do:

```py
assert set(dataset['target']) == set(dataset_shuffle['target'])
```

Reply:
yes this is less error-prone even if we added the random state.